### PR TITLE
[T-6006][17.0][ADD] purchase_split_wizard and purchase_advance_invoice modules

### PR DESCRIPTION
**Pull request by Tisho99**

### CODE REVIEW

**Review by luis-ron — APPROVED** [review]
Functional Review: LGTM 👍🏻

File: purchase_split_wizard/README.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+.. |badge3| image:: https://img.shields.io/badge/github-sygel--technology%2Fsy--purchase--workflow-lightgray.png?logo=github
+    :target: https://github.com/sygel-technology/sy-purchase-workflow/tree/16.0/purchase_split_wizard
+    :alt: sygel-technology/sy-purchase-workflow
+
+|badge1| |badge2| |badge3|
+
+This module adds a wizard that allows you to split a purchase into
+multiple ones.
+
+The split method has one partcularity, when you split a purchasethe
```
Comment (manuelregidor) [inline purchase_split_wizard/README.rst]: I don't get this bit.

File: purchase_split_wizard/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Purchase Split Wizard",
+    "summary": "Wizard to split one purchase order in two purchase orders",
```
Comment (manuelregidor) [inline purchase_split_wizard/__manifest__.py]: This is not accurate. You can split a purchase order into as many purchase orders as you want, right?

File: purchase_split_wizard/i18n/es.po [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+#. module: purchase_split_wizard
+#: model:ir.model.fields,field_description:purchase_split_wizard.field_purchase_split_wizard__currency_id
+msgid "Currency"
+msgstr "Moneda"
+
+#. module: purchase_split_wizard
+#: model:ir.model.fields,field_description:purchase_split_wizard.field_purchase_line_split_wizard__display_name
+#: model:ir.model.fields,field_description:purchase_split_wizard.field_purchase_split_wizard__display_name
+msgid "Display Name"
+msgstr "Nombre mostrado\t"
```
Comment (manuelregidor) [inline purchase_split_wizard/i18n/es.po]: Do you need "\t" in this translation?

File: purchase_split_wizard/models/purchase_order.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            raise exceptions.ValidationError(
+                _("You can not confirm splitted purchases")
+            )
+        return super().button_confirm()
+
+    def unlink(self):
+        parent_ids = self.mapped("splitted_purchase_parent_id")
+        res = super().unlink()
+        for rec in parent_ids.filtered(
+            lambda p: not any(p.splitted_purchase_children_ids)
```
Comment (manuelregidor) [inline purchase_split_wizard/models/purchase_order.py]: Why not using not lambda p: not p.splitted_purchase_children_ids?

File: purchase_split_wizard/views/purchase_order_views.xml [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            </xpath>
+            <xpath expr="//tree/field[@name='product_qty']" position="after">
+                <field name="split_status" invisible="1" />
+                <field
+                    name="qty_splitted_total"
+                    optional="hide"
+                    attrs="{'column_invisible': [('parent.split_status', 'in', ['none', 'children'])]}"
+                />
+            </xpath>
+            <!-- CONSISTENCY RESTRICTIONS-->
```
Comment (manuelregidor) [inline purchase_split_wizard/views/purchase_order_views.xml]: Do you need to keep this?

File: purchase_split_wizard/wizards/purchase_split_wizard.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+    _description = "Wizard to split purchases"
+
+    purchase_origin_id = fields.Many2one(
+        string="Purchase", comodel_name="purchase.order", readonly=True
+    )
+    company_id = fields.Many2one(
+        string="Company",
+        comodel_name="res.company",
+        required=True,
+        default=lambda self: self.env.company.id,
```
Comment (manuelregidor) [inline purchase_split_wizard/wizards/purchase_split_wizard.py]: I think you should use related="purchase_origin_id.company_id". This field is used in some domains and you have to make sure that those domains take the elements from the purchase's company, not the current one, right?

File: purchase_order_archive_draft/README.rst [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+.. |badge2| image:: https://img.shields.io/badge/licence-AGPL--3-blue.png
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-sygel--technology%2Fsy--purchase--workflow-lightgray.png?logo=github
+    :target: https://github.com/sygel-technology/sy-purchase-workflow/tree/16.0/purchase_order_archive_draft
+    :alt: sygel-technology/sy-purchase-workflow
+
+|badge1| |badge2| |badge3|
+
+This module extends the functionality of purchase_order_archive_draft to
```
Comment (manuelregidor) [inline purchase_order_archive_draft/README.rst]: I think you mean purchase_order_archive

File: purchase_split_wizard_deposit/wizards/purchase_split_wizard.py [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+        purchase_id = self.env["purchase.order"].browse(
+            res.get("purchase_origin_id")
+            or self.env.context.get("default_purchase_origin_id")
+        )
+        splitteable_advance_lines = purchase_id.order_line.filtered(
+            lambda li: ((li.qty_splittable or li.display_type) and li.is_deposit)
+        )
+        res.update(
+            {
+                "split_advances": any(splitteable_advance_lines),
```
Comment (manuelregidor) [inline purchase_split_wizard_deposit/wizards/purchase_split_wizard.py:38]: Using just splitteable_advance_lines doesn't work?

File: purchase_split_wizard_deposit/wizards/purchase_split_wizard.py [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+            rec.max_amount = rec.purchase_line_origin_id.qty_splittable
+
+    @api.constrains("amount", "max_amount")
+    def _check_max_amount(self):
+        for rec in self:
+            if not rec.display_type and rec.amount > rec.max_amount:
+                raise exceptions.ValidationError(
+                    _(
+                        "The amount can not be higher than the max amount.\n"
+                        f"- line: {rec.name}\n"
```
Comment (manuelregidor) [inline purchase_split_wizard_deposit/wizards/purchase_split_wizard.py]: I think that using f functions prevent strings from being translatable. Note that this string in not in the translations file.

**Review by manuelregidor — CHANGES_REQUESTED** [review]

File: purchase_split_wizard_deposit/wizards/purchase_split_wizard.py (hunk 9 above)
Comment (Tisho99) [reply purchase_split_wizard_deposit/wizards/purchase_split_wizard.py]: Solved!

File: purchase_split_wizard_deposit/wizards/purchase_split_wizard.py (hunk 8 above)
Comment (Tisho99) [reply purchase_split_wizard_deposit/wizards/purchase_split_wizard.py:38]: It does work, but the any converts it to boolean and i like it :3

File: purchase_split_wizard/views/purchase_order_views.xml (hunk 5 above)
Comment (Tisho99) [reply purchase_split_wizard/views/purchase_order_views.xml]: no

File: purchase_split_wizard/models/purchase_order.py (hunk 4 above)
Comment (Tisho99) [reply purchase_split_wizard/models/purchase_order.py]: Removed

File: purchase_split_wizard/i18n/es.po (hunk 3 above)
Comment (Tisho99) [reply purchase_split_wizard/i18n/es.po]: no

File: purchase_split_wizard/__manifest__.py (hunk 2 above)
Comment (Tisho99) [reply purchase_split_wizard/__manifest__.py]: Changed

File: purchase_split_wizard/README.rst (hunk 1 above)
Comment (Tisho99) [reply purchase_split_wizard/README.rst]: I have updated the readme and explained it a bit more

**Comment by Tisho99** [comment]
@manuel-florido do you know what is this pre-commit error? Dont look the tests, solving the tests is still not necesary

**Review by manuelregidor — APPROVED** [review]
Technical review. LGTM.

**Comment by manuel-florido** [comment]
@Tisho99 

Your branch is outdated, rebase needed

**Comment by Tisho99** [comment]
@manuelregidor can you review again?

**Comment by Tisho99** [comment]
@ValentinVinagre can you merge?

**Pull request merged**